upload/views/helpers.py

```diff
@@ -5,22 +5,28 @@
 
 
 def get_repository_from_string(
-    service: Service, repo_identifier: str
-) -> typing.Optional[Repository]:
+    service: Service, repo_identifier: str, include_owner: bool = False
+) -> typing.Optional[Repository] | tuple[Repository | None, Owner | None]:
```

Comment:
[nit] 
this `include_owner` is unnecessarily complicating this function.

And it looks like it was added for the benefit of `get_repository_and_owner_from_string`. We have a `_get_owner_from_string` already.

I'd personally think the extra logic should be moved to `get_repository_and_owner_from_string`. Have that make use of the 2 other funcitons.

Reply:
agree! updating...